Mathlib/Algebra/Polynomial/Degree/Lemmas.lean

```diff
@@ -324,6 +343,13 @@ lemma natDegree_eq_one : p.natDegree = 1 ↔ ∃ a ≠ 0, ∃ b, C a * X + C b =
   · rintro ⟨a, ha, b, rfl⟩
     simp [ha]
 
+theorem subsingleton_isRoot_of_natDegree_eq_one [IsLeftCancelMulZero R] [IsRightCancelAdd R]
+    (h : p.natDegree = 1) : { x | IsRoot p x }.Subsingleton := by
+  intro x hx y hy
+  have ⟨a, ha, b, hp⟩ := Polynomial.natDegree_eq_one.mp h
+  simp [← hp] at hx hy
```

Comment:
```suggestion
  obtain ⟨a, ha, b, rfl⟩ := Polynomial.natDegree_eq_one.mp h
  simp only [IsRoot.def, eval_add, eval_mul_X, eval_C, Set.mem_setOf_eq] at hx hy
```
You have a non-terminal simp, i.e. your simp does not close a goal, so you should convert that to `simp?`, which we call "squeezing the simp"; and then when you click on the result, you will get the line I have suggested.